### ISSUE TITLE
Bug in RightDivide

### ISSUE THREAD

**Issue by sebastianpos** [homalg-project/homalg_project#138]
The following sequence of commands leads to an error:
```
LoadPackage( "MatricesForHomalg" );
QQ := HomalgFieldOfRationals( );
A := HomalgMatrix( [ [ 1, 0 ], [ 0, 1 ] ], 2, 2, QQ );
B := HomalgMatrix( [ [ 1, 0 ], [ 0, 1 ] ], 2, 2, QQ );
R := RightDivide( A, B );
Display( R );
```

**Comment by sebasguts** [comment]
This was caused by 1d81d423afc93f5dcad6e2b3ea3d8f9d3af10 . Please note that the change produces a different output. Maybe the new output is simply wrong. I reverted the commit.

**Comment by mohamed-barakat** [comment]
closed by the reverse commit 13193f8
